### PR TITLE
Validate that the metadata is always a dict

### LINKED ISSUE #428: Validate that the metadata is always a dict
### Problem Description
The diagnostic and quality reports always expect that the metadata is a _dictionary_. This can be confusing because in other libraries (such as the SDV), the metadata is actually its own object.

### Expected behavior
If I don't pass in a dictionary into a report, provide a helpful error message.

```python
from sdmetrics.reports.single_table import DiagnosticReport

report = DiagnosticReport()
report.generate(real_data, synthetic_data, metadata_object)
```

```
InvalidInputError: The provided metadata is not a dictionary.
```

### Additional context
We cannot use metadata objects in SDMetrics because we are not using or importing the SDV.

### PR DESCRIPTION
**Pull request by R-Palazzo**
Resolve #428

### CODE REVIEW

**Comment by npatki** [comment]
@R-Palazzo we should not be accepting the actual metadata object. We should only accept dictionaries.

I would avoid doing any conversions in SDMetrics and stick to the original issue.

File: sdmetrics/reports/base_report.py [hunk 1]
```diff
@@ -55,14 +55,25 @@ def validate(self, real_data, synthetic_data, metadata):
...
 
     def _handle_results(self, verbose):
         raise NotImplementedError
 
+    @staticmethod
+    def _convert_metadata(metadata):
+        """If the metadta is not a dict, try to convert it."""
+        if not isinstance(metadata, dict):
+            try:
+                metadata = metadata.to_dict()
```
Comment (frances-h) [inline sdmetrics/reports/base_report.py:68]: Is this something we actually want to support? Other objects might implement a `to_dict` method which might lead to weird issues down the road (i.e. I'm pretty sure you can call `to_dict` on a pandas dataframe). Maybe @npatki has thoughts?

Comment (npatki) [reply sdmetrics/reports/base_report.py:68]: @frances-h agreed. I think we should stick to the issue's description, which states that we should only accept dictionaries. We do not intend to convert to a dictionary on the user's behalf.

**Review by amontanez24 — APPROVED** [review]
LGTM!

**Pull request merged**